### PR TITLE
CLN/REF: Refactor min_periods calculation in rolling

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/window/test_rolling.py [hunk 1]
```diff
@@ -498,7 +498,7 @@ def test_rolling_count_default_min_periods_with_null_values(constructor):
                 ({"A": [2, 3], "B": [5, 6]}, [1, 2]),
             ],
             2,
-            3,
+            2,
```
Comment (mroeschke) [inline pandas/tests/window/test_rolling.py:501]: Related to the whatsnew entry: This should not have passed previously.

Comment (jreback) [reply pandas/tests/window/test_rolling.py:501]: what whatsnew entry?

Comment (mroeschke) [reply pandas/tests/window/test_rolling.py:501]: Oops, though I added it is previously. Should be there now.

**Comment by jreback** [comment]
cool. was there a referenced issue?

**Comment by mroeschke** [comment]
No, just discovered independently

**Pull request merged**